server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -661,9 +664,8 @@ private static void parseNonDynamicArray(
                 && (objectMapper.storeArraySource() || objectMapper.dynamic == ObjectMapper.Dynamic.RUNTIME);
             boolean fieldWithFallbackSyntheticSource = mapper instanceof FieldMapper fieldMapper
                 && fieldMapper.syntheticSourceMode() == FieldMapper.SyntheticSourceMode.FALLBACK;
-            boolean nestedObject = mapper instanceof NestedObjectMapper;
```

Comment:
Should we keep this, along with a check on `storeArraySource`?

Reply:
The previous check (`objectRequiresStoringSource`) should be enough since `NestedObjectMapper` is an instance of `ObjectMapper`.

Reply:
Indeed, since we now use the same param.